docs(2022 12 08): add docs for folder structure and url paths (#317)
dhis2/maintenance-app-beta: docs/folder-structure.md

> # Folder structure

Reviewer: Consider moving this file to `docs/architecture/folder-structure.md`

## Folders

### `app`

> This folder will contain everything that's related to the global part of the

Reviewer: ```suggestion
This folder contains everything that's related to the global part of the
```

app, e.g. the app layout, the navigation or setting up providers

### `components`

This folder contains all components that are shared between multiple pages
and/or the app module

### `utils`

This contains all non-component functionality by features that's shared between
pages and/or the app

### `pages`

> Every page will have its own folder inside of here, so we can easily colocate

Reviewer: ```suggestion
Every page has its own folder inside of here, so we can easily colocate
```
